src/modules/collection/sagas.ts

```diff
@@ -226,3 +236,50 @@ function* handleMintColectionItems(action: MintCollectionItemsRequestAction) {
 function* handleConnectWalletSuccess() {
   yield put(fetchCollectionsRequest())
 }
+
+function* handleTransactionSuccess(action: FetchTransactionSuccessAction) {
+  const transaction = action.payload.transaction
+
+  try {
+    switch (transaction.actionType) {
+      case PUBLISH_COLLECTION_SUCCESS: {
+        // We re-fetch the collection from the store to get the updated version
+        const collectionId = transaction.payload.collection.id
+        const collection = yield select(state => getCollection(state, collectionId))
+        const items = yield select(state => getCollectionItems(state, collectionId))
+
+        for (const item of items) {
```

Comment:
I'd filter out the items that are already `inCatalyst` to prevent sending an unnecessary upload request

Reply:
Thought about it but wasn't sure, gotcha!